.github/workflows/preview.yaml

```diff
@@ -0,0 +1,43 @@
+name: Preview Deployment
+env:
+  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+on:
+  push:
+    branches-ignore:
+      - main
```

Comment:
Suggestion: this workflow will run for a new push on every branch which may not be intended. It might be appropriate to constrain this to just pull requests to the `main` branch.